### DISCUSSION TITLE
How to replicate StructuredDataClassifier and Regressor functionality in 2.0?

### DISCUSSION THREAD

**Discussion by adunega** [keras-team/autokeras#1914] [Q&A] [answered]
Hi there, huge fan of AutoKeras.  It's been a tremendous tool for my financial research.  I just noticed in the new release of 2.0 that the primary functions I was using were dropped from the code (StrucutredDataClassifier and StructuredDataRegressor).  

I'm curious on the reason for dropping those, but more importantly do you have any recommendations/advice on another tool I should use for replacing that functionality?  Can I rebuild it myself using AutoModel or should I just fork AK 1.1 if I want to keep using those functions?

Thanks in advance for your help.

**Comment by haifeng-jin** [comment, marked as answer]
Thanks, @adunega !

Yes, it is removed.
The main reason is the search space is limited to neural networks, which may not be as good as other models for structured data.
It is also adding a lot of maintenance overhead.

The feature encoding part is also not good enough since AutoML cannot understand the features but only brute-forcely search it.

I suggest you to do the feature encoding for all categorical features by yourself and use it with a normal `Input()` and `RegressionHead()` or `ClassificationHead()` with the IO API in AutoKeras.

Or you can just use [XGBoost](https://xgboost.readthedocs.io/en/stable/), which may be better for structured data tasks.

**Reply by adunega** [reply]
Thanks for the explanation and suggestions!

**Reply by BradKML** [reply]
I would like to see documentation for tabular data to come back (classification and regression), since most articles recommended using `StructuredDataRegressor` which does not exist anymore in the newer libraries https://github.com/keras-team/autokeras/issues/1915
